lib/Test2/Harness/Collector.pm

```diff
@@ -214,16 +218,49 @@ sub process_tasks {
     return $count;
 }
 
+sub send_backed_up {
+    my $self = shift;
+    return if $self->{+BACKED_UP}++;
+
+    # This is an unlikely code path. If we're here, it means the last loop couldn't process any results.
+    my $e = $self->_harness_event(0, undef, time, info => [{details => <<"    EOT", tag => "INTERNAL", debug => 1, important => 1}]);
+*** THIS IS NOT FATAL ***
+
+ * The collector has reached the maximum number of concurrent jobs to process.
+ * Testing will continue, but some tests may be running or even complete before they are rendered.
+ * All tests and events will eventually be displayed, and your final results will not be effected.
+
+Set a higher --max-open-jobs collector setting to prevent this problem in the
+future, but be advised that could result in too many open filehandles on some
+systems.
+
+This message will only be shown once.
+    EOT
+
+    $self->{+ACTION}->($e);
+    return;
+}
+
 sub jobs {
     my $self = shift;
 
     my $jobs = $self->{+JOBS} //= {};
 
     return $jobs if $self->{+JOBS_DONE};
 
+    # Don't monitor more than 'max_open_jobs' or we might have too many open file handles and crash
+    # Max open files handles on a process applies. Usually this is 1024 so we
+    # can't have everything open at once when we're behind.
+    my $max_open_jobs = $self->settings->collector->max_open_jobs // 1024;
+    my $additional_jobs_to_parse = $max_open_jobs - keys %$jobs;
+    if($additional_jobs_to_parse <= 0) {
+        $self->send_backed_up;
```

Comment:
This should never happen. If it does, it means an entire processing loop happened and no jobs were processed to completion. Given there are 300 jobs, that'd be bad or you passed a REALLY big number to -j.

Reply:
It happened in my testing when I set -j to 2 and set --max-open-jobs to 1. Not everyone will be using the defaults.

Reply:
Sure but I would argue it's a bug to set --max-open-jobs  to less than -j and maybe we should warn on that?